File: nozomi/src/types/album.ts
import { ArtistPartial } from "./artistPartial"

export interface Album {
    id: number
    name: string
    art: string[]
    year: any
    created_at: number[]
    updated_at: any
    artist: Artist
    tracks: Track[]
  }
  
  export interface Artist {
    id: number
    name: string
    picture: string
  }
  
  export interface Track {
    id: number
    name: string
    album_artist: number
    artists: ArtistPartial[]
    plays: number
    duration: number
    liked: boolean
    last_play: any
    year: any
    number: number
    disc: number
    lossless: boolean
    created_at: number[]
    updated_at: any
    album: number
    album_name: string
    artist_name: string
  }
  

File: muse/muse/Models/Album.swift
//
//  Album.swift
//  muse
//
//  Created by Natalie on 3/24/26.
//

import Foundation

struct AlbumPartial: Codable, Identifiable, Hashable {
    let id: Int
    let name: String
    let art: [String]
    let year: Int?
    let count: Int?
    let artist: ArtistPartial?

    var primaryArtUrl: String? {
        art.first
    }

    func hash(into hasher: inout Hasher) {
        hasher.combine(id)
    }

    static func == (lhs: AlbumPartial, rhs: AlbumPartial) -> Bool {
        lhs.id == rhs.id
    }
}

struct Album: Codable, Identifiable, Hashable {
    let id: Int
    let name: String
    let art: [String]
    let year: Int?
    let createdAt: Date
    let updatedAt: Date?
    let artist: ArtistPartial
    let tracks: [Track]?

    var primaryArtUrl: String? {
        art.first
    }

    func hash(into hasher: inout Hasher) {
        hasher.combine(id)
    }

    static func == (lhs: Album, rhs: Album) -> Bool {
        lhs.id == rhs.id
    }
}

struct AllAlbumsPartial: Codable {
    let albums: [AlbumPartial]
    let limit: Int
    let offset: Int
}
